docs/Performance/promise.md

```diff
@@ -0,0 +1,195 @@
+
+```js
+var model = that.getView().getModel();
+model.read("/MyEntitySet('12345')", {
+  success: function (data, response) {
+    // success handler
+  },
+  error: function (error) {
+    // error handler
+  }
+});
+```
+
+### Promisified OData call
+
+A wrapped version of the basic OData call.
+
+The key is given as parameter and the wrapping method returns a promise. The resolve / reject methods are used in the success / error callbacks to resolve or reject the Promise.
+
+As the resolve method takes only one parameter, data and response are passed as object.
+
+```js
+function readMyEntity(id) {
+  return new Promise(function (resolve, reject) {
+    var model = that.getView().getModel();
+    model.read("/MyEntitySet('" + id + "')", {
+      success: function (data, response) {
+        resolve({ data: data, response: response });
+      },
+      error: function (error) {
+        reject(error);
+      }
+    });
+  });
+}
+```
+
+The success and error handling can now be done after invoking the method.
+
+```js
+readMyEntity(12345)
+  .then(function (result) {
+    // success handler
+    console.log(result.data, result.response);
+  })
+  .catch(function (error) {
+    // error handler
+    console.error(error);
+  });
+```
+
+### Refactor the Promisified OData call
+
+Now, let's make the OData call more robust for everyday usage:
+
+* hand over the whole entity as object and construct the key from its data
+* provide an option to add additional data like filters to the call (mParameters as described in the [documentation](https://ui5.sap.com/#/api/sap.ui.model.odata.v2.ODataModel%23methods/read))
+* make sure its not possible to provide success / error callbacks with the parameters object as this would break the promise functionality
+
+The UI5 framework detects when the same OData call is done multiple times. It does only one request and returns the same result object for every call. If this result object is being changed, it changes for all calls. To prevent this from happening, the result is being copied with Object.assign()
+
+If you plan to use this with IE11, make sure to provide the [Object.assign Polyfill](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#polyfill).
+
+```js
+/** 
+ * Returns the result of the OData call
+ * @param {object} payload - The payload which equals to the entity
+ * @param {object} parameters - The parameters like Filters added to the OData call
+ * @returns {Promise} Promise object represents result of the OData call
+ */
+function readMyEntity(payload, parameters) {
+  var model = this.getView().getModel();
+  parameters = parameters || {};
+  return model.metadataLoaded()
+    .then(function () {
+      return new Promise(function (resolve, reject) {
+        var key = model.createKey("/MyEntitySet", payload)
+        // prevent success / error callbacks to be overwritten
+        var params = Object.assign({}, parameters,
```

Comment:
As you've noted, `Object.assign` doesn't work in IE11 and a polyfill is required. So probably explicitly change to `JSON.parse(JSON.stringify(parameters))`?

Reply:
added this as a note